tests/expose_test.go

```diff
@@ -331,89 +615,166 @@ var _ = Describe("[rfe_id:253][crit:medium][vendor:cnv-qe@redhat.com][level:comp
-			err = virtctl()
-			Expect(err).ToNot(HaveOccurred())
+			virtctl := tests.NewRepeatableVirtctlCommand("start", "--namespace", namespace, name)
+			Expect(virtctl()).To(Succeed(), "should succeed starting a VMI via `virtctl start ...`")
 
 			By("Getting the status of the VMI")
 			Eventually(func() bool {
-				vm, err = virtClient.VirtualMachine(vm.Namespace).Get(vm.Name, &k8smetav1.GetOptions{})
+				vm, err = virtClient.VirtualMachine(namespace).Get(name, &k8smetav1.GetOptions{})
 				Expect(err).ToNot(HaveOccurred())
 				return vm.Status.Ready
 			}, 120*time.Second, 1*time.Second).Should(BeTrue())
 
 			By("Getting the running VMI")
 			var vmi *v1.VirtualMachineInstance
 			Eventually(func() bool {
-				vmi, err = virtClient.VirtualMachineInstance(vm.Namespace).Get(vm.Name, &k8smetav1.GetOptions{})
+				vmi, err = virtClient.VirtualMachineInstance(namespace).Get(name, &k8smetav1.GetOptions{})
 				Expect(err).ToNot(HaveOccurred())
 				return vmi.Status.Phase == v1.Running
 			}, 120*time.Second, 1*time.Second).Should(BeTrue())
 
-			tests.GenerateHelloWorldServer(vmi, testPort, "tcp")
-		})
+			return vmi
+		}
+
+		createStoppedVM := func(virtClient kubecli.KubevirtClient, namespace string) error {
+			By("Creating an VM object")
+			template := newLabeledVMI("vm", virtClient, false)
+			vm = tests.NewRandomVirtualMachine(template, false)
+
+			By("Creating the VM")
+			vm, err = virtClient.VirtualMachine(namespace).Create(vm)
+			return err
+		}
 
 		Context("Expose a VM as a ClusterIP service.", func() {
-			It("[test_id:1538][label:masquerade_binding_connectivity]Connect to ClusterIP service that was set when VM was offline.", func() {
+			tests.BeforeAll(func() {
+				tests.BeforeTestCleanup()
+			})
+
+			BeforeEach(func() {
+				Expect(createStoppedVM(virtClient, tests.NamespaceTestDefault)).To(Succeed(), "should have a stopped VM.")
+			})
+
+			BeforeEach(func() {
+				vmExposeArgs = []string{
+					expose.COMMAND_EXPOSE,
+					"virtualmachine", "--namespace", vm.GetNamespace(), vm.GetName(),
+					"--port", servicePort, "--name", serviceName, "--target-port", strconv.Itoa(testPort),
+				}
+			})
+
+			BeforeEach(func() {
+				jobCleanupFuncs = nil
+			})
+
+			BeforeEach(func() {
+				serviceCleanupFunc = nil
+			})
+
+			AfterEach(func() {
+				Expect(jobCleanupFuncs).NotTo(BeNil(), "a successful test must have stored a way to delete the batchv1.Job entity")
+				Expect(cleanupJobs(jobCleanupFuncs)).To(BeEmpty(), "should be able to delete the multiple k8sv1.`Job` entities")
+			})
+
+			AfterEach(func() {
+				Expect(serviceCleanupFunc).NotTo(BeNil(), "a successful test must have stored a way to delete the k8sv1.Service entity")
+				Expect(serviceCleanupFunc()).To(Succeed(), "should be able to delete the k8sv1.Service entity")
+			})
+
+			AfterEach(func() {
+				if vm != nil {
+					By("Calling the stop command on the running VMI")
+					Expect(vm).NotTo(BeNil(), "should have a VM running so we can get rid of the VMI")
```

Comment:
didnt we check it on line 691 ?
either leftovers or you want other check (or i am wrong)